performance/query_processing.rst

```diff
@@ -61,6 +61,28 @@ The task tracker executor is well suited for long running, complex data warehous
 
 Each task tracker daemon on the workers also makes sure to execute at most citus.max_running_tasks_per_node concurrently. This concurrency limit helps in avoiding disk I/O contention when queries are not served from memory. The task tracker executor is designed to efficiently handle complex queries which require repartitioning and shuffling intermediate data among workers.
 
+Subquery/CTE Push-Pull Execution
+--------------------------------
+
+If necessary Citus can push results from subqueries and CTEs into short-lived :ref:`reference_tables` for use by an outer query. This allows Citus to support a greater variety of SQL constructs, and even mix executor types between a query and its subqueries.
```

Comment:
Maybe note that "For CTE's and Subqueries which cannot be flattened or otherwise pushed down, Citus will recursively plan and execute these independently. The results of these are gathered on the coordinator and then pushed back to the workers in order to join or combine the results".

As noted in a separate comment, these aren't reference tables. So we should remove the link here, and also the text. We push them as files and read from them directly. Conceptually they are like reference tables, but if we call them that, users might get confused about what they could do with it.